.github/workflows/main.yml

```diff
@@ -25,6 +25,9 @@ jobs:
     steps:
       - uses: actions/checkout@master
 
+      - name: Ensure the terraform files are semantically valid
+        run: terraform validate
```

Comment:
It doesn't do anything here at the top-level, it needs to be run in a Terraform configuration directory.